### DISCUSSION TITLE
Make focal work on 3d-arrays

### DISCUSSION THREAD

**Discussion by sebiegli** [makepath/xarray-spatial#708] [Ideas]
I'd like to use the focal-functionalities of xarray-spatial to calculate some statistics for a raster with multiple bands (in my case R, G and B). Unfortunately, so far, focal seems to be limited to 2d-arrays, eg. when handing over a RGB-3d-array to _focal.apply()_ I get this error message:

`ValueError: `raster` must be 2D`

Is there a way of calculating focal statistics on multiple bands? Or are there plans to include this functionality in the future?

**Comment by brendancol** [comment]
@sebiegli sorry for the delayed response.  The was just merged in #924
